src/AgentHeader.php

```diff
@@ -107,6 +111,15 @@ public static function buildAgentHeader($headerInfo)
             : $apiCoreVersion;
         $metricsHeaders['rest'] = $restVersion;
 
+        if (isset($headerInfo['customKeyValues'])) {
```

Comment:
Since these custom values are fundamentally different from the existing `$headerInfo` values, I would prefer to add a separate argument to `buildAgentHeader`:

```php
public static function buildAgentHeader($headerInfo, array $additionalValues = [])
```